src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -232,7 +234,7 @@ public void log(String message) {
         throw new DeployException("Deployment hosts of type " + configuration.getClass() + " are unsupported");
     }
 
-    public static class FingerprintDecorator implements Callable<DeployedApplicationLocation, IOException> {
+    public static class FingerprintDecorator extends MasterToSlaveCallable<DeployedApplicationLocation, IOException> {
```

Comment:
> Is that it?

Yes.

> if you think this is something to improve or something that you think that is required

I think fixing this is required. Certainly the PR as it currently stands is incorrect. Anyway it should be a pretty small refactoring; I am happy to offer a suggested commit if you are unsure.